File: Modules/WorkFlow/Status/StatusColumns.cs
namespace WorkFlow.Columns;

[ColumnsScript("WorkFlow.Status")]
[BasedOnRow(typeof(StatusRow), CheckNames = true)]
public class StatusColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink, Width(250)]
    public string Name { get; set; }
    [Width(150)]
    public DateTime UpdateDate { get; set; }
}

File: Modules/ServerTypes/WorkFlow/StatusColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { StatusRow } from "./StatusRow";

export interface StatusColumns {
    Id: Column<StatusRow>;
    Name: Column<StatusRow>;
    UpdateDate: Column<StatusRow>;
}

export class StatusColumns extends ColumnsBase<StatusRow> {
    static readonly columnsKey = 'WorkFlow.Status';
    static readonly Fields = fieldsProxy<StatusColumns>();
}